docs/reference/indices/update-settings.asciidoc

```diff
@@ -60,6 +60,16 @@ include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=index-ignore-unavailab
 (Optional, Boolean) If `true`, existing index settings remain unchanged.
 Defaults to `false`.
 
+`reopen`::
+(Optional, Boolean) If `true`, then any static settings that would ordinarily only
+be updated on closed indices will be updated by automatically closing and reopening
+the affected indices. If `false`, attempts to update static settings on open indices
+will fail. Defaults to `false`.
+
+WARNING: Changing index settings on an automatically closed index using the `reopen`
+parameter could result in incorrect settings that are impossible to rectify without
+deleting and recreating the index.
```

Comment:
Hm that sounds scary. What does it mean exactly? Are there circumstances where you cannot just revert the settings change?

Reply:
I copied that from the documentation about changing settings on closed indices. If it is true there, I assume it is true here. I don't have any examples though. If it's not true, do we even need this explicit setting? When would a person ever choose _not_ to update the setting?

Reply:
Here's the existing warning: https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html#index-modules. Other people have asked the same question about which settings this warning is referring to: https://discuss.elastic.co/t/changing-index-settings-when-closed-can-corrupt-an-index/326705

Reply:
Huh TIL again. That was added 8 years ago in https://github.com/elastic/elasticsearch/commit/f123a53d7258349a171e47a35f4581899d8fa776 and I don't see any associated commentary about why. I don't think it's correct, at least, I hope it's not, and I haven't heard of anyone encountering this situation in a support case either. Let's drop it here, and would you also open an issue to start a discussion about dropping it from the other index settings docs?

Reply:
> If it's not true, do we even need this explicit parameter?

Reopening an index triggers some unavailability while the shards re-initialize. That may be a source of surprise (and disappointment) to users so it needs to be an opt-in thing. We should probably say that explicitly in these docs just to make sure.

Reply:
What do you think of the updated less scary wording? I'm trying to avoid use of the word "shards" in the public-facing documentation.

Reply:
Ticket for the warning in the index modules docs: https://github.com/elastic/elasticsearch/issues/101851